index.js

```diff
@@ -16,58 +18,155 @@ const QUESTIONS = [{
+    return res.status(401).send("Invalid user email-ID or  password");
+  }
+
   // Also ensure that the password is the same
+  // password----checking
+  if(user.password !== password){
+    // incorrect!
+    return res.status(401).send("Invalid user email-ID or  password");
+  }
 
 
-  // If the password is the same, return back 200 status code to the client
-  // Also send back a token (any random string will do for now)
-  // If the password is not the same, return back 401 status code to the client
 
+  //if the email and password is correct, then a token is created and it is send back to the client
+  const token = "random token";
 
-  res.send('Hello World from route 2!')
-})
+  res.status(200).json({ token })
+});
 
 app.get('/questions', function(req, res) {
 
   //return the user all the questions in the QUESTIONS array
-  res.send("Hello World from route 3!")
-})
+  res.json(QUESTIONS);
+});
 
 app.get("/submissions", function(req, res) {
+    //problemid is add
+    const problemId = req.query.problemId;
+
+    //find all the submission for the given problemId
+    const problemSubmission = SUBMISSION.filter(submission => submission.problemId === problemId)
+
    // return the users submissions for this problem
-  res.send("Hello World from route 4!")
+  res.status(200).json(problemSubmission);
 });
 
 
 app.post("/submissions", function(req, res) {
-   // let the user submit a problem, randomly accept or reject the solution
-   // Store the submission in the SUBMISSION array above
-  res.send("Hello World from route 4!")
+    const { problemId, solution} = req.body;
+
+    //Randomly accept or reject the soltuion
+    const isAccepted = Math.random() < 0.5;
+
+
+    //create a new submission object with problemId, soultion along with isAccepted
+    const submission = {
+      problemId,
+      solution,
+      isAccepted
+    };
+
+    //store the submission in the SUBMISSION array
+    SUBMISSION.push(submission);
+
+
+   //Return a response whether the solution is accepted or not
+
+    const status = isAccepted ? "accepted" : "rejected";
+  //successful status send
+  res.status(200).json({ status })
 });
 
 // leaving as hard todos
 // Create a route that lets an admin add a new problem
+
+const jwt = require('jsonwebtoken');
+const SECRET_KEY = 'MERA_KEY_HAI__TUMHE_KYU_BATAOO...';
+
+const PROBLEMS = [];
+
+let lastId = QUESTIONS.length;
+app.post('/addproblems',function (req, res){
+  const { title, description, testCases } = req.body;
+
+  //check if the user is authenticated and authorised
+  const authHeader = req.headers.authorization;
+  if(!authHeader || !authHeader.startsWith('Bearer ')){
+    return res.status(401).json({ message: 'Unauthorized'});
+  }
+
+  const token = authHeader.split(' ')[1];
+  let decodedToken;
+
+  try{
+    decodedToken = jwt.verify(token, SECRET_KEY);
+  }catch (error) {
+      return res.status(401).json({message : 'unauthorized'});
+  }
+
+  if(!decodedToken || decodedToken !== 'admin'){
```

Comment:
It's good to have a user role stored in you data store and since you have used JWT approach for tokens you can add role 'admin' as payload while signing JWT and once the JWT verification is done you can check for `role` field in JWT payload provided by user